FogCreek/Glitch-Community
# Superb guide: Optimizely refactors + support for ssr based on user id

## Links
https://superb-guide.glitch.me/
https://app.clubhouse.io/glitch/story/1593/a-b-testing-with-optimizely

## Screenshots
![image](https://user-images.githubusercontent.com/8932174/68044773-6590ac80-fcae-11e9-8803-63d2a452b3e3.png)

## Changes:
- Track a new cookie `optimizely-id` which is set to the current user id if you are logged in. The server assigns a random number if you don't already have an id
- SSR caches based on the current user's feature flags (so it doesn't have to render every request)
- Added a table to the secret page that shows what flags are active for your user id

## How To Test:
Try loading up the site and check what features are enabled, they should stay across reloads. Sign in, and notice that the enabled features might have changed. Sign out, they should stay the same as when you were signed in.

## Feedback I'm looking for:
The analytics feature is based on the team id, but the secret page uses your user id. Should it do anything to indicate that?

## Things left to do before deploying:
- I need to get the `optimizely-id` cookie whitelisted in cloudfront
- (not blocking deploy) there should be a way to force enable/disable features
